python/trifinger_simulation/gym_wrapper/__init__.py

```diff
@@ -1,15 +1,21 @@
 from gym.envs.registration import register
+from trifinger_simulation.gym_wrapper.envs import trifinger_cube_env
 
 register(
-    id="reach-v0",
+    id="TriFingerReach-v0",
     entry_point="trifinger_simulation.gym_wrapper.envs.trifinger_reach:TriFingerReach",
 )
 register(
-    id="push-v0",
+    id="TriFingerPush-v0",
     entry_point="trifinger_simulation.gym_wrapper.envs.trifinger_push:TriFingerPush",
 )
 
-register(
-    id="real_robot_challenge_phase_1-v1",
-    entry_point="trifinger_simulation.gym_wrapper.envs.cube_env:CubeEnv",
-)
+for difficulty in [1, 2, 3, 4]:
+    initializer = trifinger_cube_env.RandomInitializer(difficulty=difficulty)
+    kwargs= {'initializer': initializer}
+    register(
+        id="TriFingerCubeDifficulty{}-v1".format(difficulty),
+        entry_point="trifinger_simulation.gym_wrapper.envs.trifinger_cube_env:TriFingerCubeEnv",
+        kwargs=kwargs,
+    )
```

Comment:
I see a potential issue here: The difficulty is communicated to the environment only through the initializer.  From what I understand, this sets a default initializer but the user can still pass a custom one when creating the env?
In this case, the difficulty in the name of the environment does not have any meaning.  The user can easily work with a different difficulty in the initializer as is specified in the name which would be very confusing.
Unless I'm overseeing something, I don't think that this is a good change as it is now.